file/pkg/model/fileshare.go

```diff
@@ -0,0 +1,107 @@
+	// the user.
+	// +readOnly
+	Id bson.ObjectId `json:"id,omitempty" bson:"_id,omitempty"`
+
+	// CreatedAt representing the server time when the object was created successfully.
+	// Now, it's represented as a time string in RFC8601 format.
+	// +readOnly
+	CreatedAt string `json:"createdAt" bson:"createdAt,omitempty"`
+
+	// UpdatedAt representing the server time when the object was updated successfully.
+	// Now, it's represented as a time string in RFC8601 format.
+	// +readOnly
+	UpdatedAt string `json:"updatedAt" bson:"updatedAt,omitempty"`
+
+	// The uuid of the project that the file share belongs to.
+	TenantId string `json:"tenantId,omitempty" bson:"tenantId,omitempty"`
+
+	// The uuid of the user that the file share belongs to.
+	// +optional
+	UserId string `json:"userId,omitempty" bson:"userId,omitempty"`
+
+	// The uuid of the backend that the file share belongs to.
+	BackendId string `json:"backendId,omitempty" bson:"backendId,omitempty"`
+
+	// The name of the backend that the file share belongs to.
+	Backend string `json:"backend,omitempty" bson:"backend,omitempty"`
+
+	// The name of the file share.
+	Name string `json:"name,omitempty" bson:"name,omitempty"`
+
+	// The description of the file share.
+	// +optional
+	Description string `json:"description,omitempty" bson:"description,omitempty"`
+
+	// The type of the file share.
+	Type string `json:"type,omitempty" bson:"type,omitempty"`
+
+	// The size of the file share requested by the user.
+	// Default unit of file share Size is GB.
+	// +optional
+	Size int64 `json:"size,omitempty" bson:"size,omitempty"`
+
+	// The location that file share belongs to.
+	// +optional
+	Region string `json:"region,omitempty bson:"region,omitempty"`
+
+	// The locality that file share belongs to.
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty bson:"availabilityZone,omitempty"`
+
+	// The status of the file share.
+	Status string `json:"status,omitempty" bson:"status,omitempty"`
+
+	// The protocol of the fileshare. e.g NFS, SMB etc.
+	// +optional
+	Protocols []string `json:"protocols,omitempty" bson:"protocols,omitempty"`
+
+	// Any tags assigned to the file share.
+	// +optional
+	Tags []Tag `json:"tags,omitempty" bson:"tags,omitempty"`
+
+	// The uuid of the snapshot from which the file share is created.
+	// +optional
+	SnapshotId string `json:"snapshotId,omitempty bson:"snapshotId,omitempty"`
+
+	// Indicates whether the file share is encrypted.
+	// +optional
+	Encrypted bool `json:"encrypted,omitempty" bson:"encrypted,omitempty"`
+
+	// EncryptionSettings that was used to protect the file share encryption.
+	// +optional
+	EncryptionSettings map[string]string `json:"encryptionSettings,omitempty bson:"encryptionSettings,omitempty"`
+
+	// Metadata should be kept until the semantics between file share and backend storage resource.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty bson:"metadata,omitempty"`
```

Comment:
What is metadata for?

Reply:
Meta Data is a placeholder for keeping cloud specific parameters like for AWS, Metadata may contain the following:
 NumberOfMountTargets: 3,
 OwnerId: "75027",
 PerformanceMode: "generalPurpose",
 SizeInBytes: {
        Timestamp: 2020-06-03 11:05:32 +0000 UTC,
        Value: 6144,
        ValueInIA: 0,
        ValueInStandard: 6144
  }
FileSystemId: "fs-4ffe7d9e",
LifeCycleState: "available",

Reply:
ok